### PR TITLE
add http api to akka-sample-cqrs-java

### PR DESCRIPTION
**Pull request by patriknw**
Corresponding for Scala was https://github.com/akka/akka-samples/pull/156

### CODE REVIEW

File: akka-sample-cqrs-java/src/main/java/sample/cqrs/ShoppingCartRoutes.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+            entity(
+              Jackson.unmarshaller(objectMapper, UpdateItem.class),
+              this::handleUpdateItem))
+        )
+      )
+    );
+  }
+
+  private Route handleAddItem(AddItem data) {
+    System.out.println("handleAddItem");
```
Comment (johanandren) [inline akka-sample-cqrs-java/src/main/java/sample/cqrs/ShoppingCartRoutes.java]: runaway println

File: akka-sample-cqrs-java/src/main/java/sample/cqrs/ShoppingCartRoutes.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    EntityRef<ShoppingCart.Command> entityRef =
+      sharding.entityRefFor(ShoppingCart.ENTITY_TYPE_KEY, data.cartId);
+    CompletionStage<ShoppingCart.Confirmation> reply =
+      entityRef.ask(replyTo -> new ShoppingCart.AddItem(data.itemId, data.quantity, replyTo), timeout);
+    return onConfirmationReply(reply);
```
Comment (johanandren) [inline akka-sample-cqrs-java/src/main/java/sample/cqrs/ShoppingCartRoutes.java]: I would have rather extracted the entity-ref + ask part into a method and then kept the HTTP routes API usage in the route tree (Like in the Java quickstart) that makes for a cleaner separation of concerns IMO

Comment (patriknw) [reply akka-sample-cqrs-java/src/main/java/sample/cqrs/ShoppingCartRoutes.java]: I think I tried to make the lambdas small, but your argument makes sense. Changed in 1fd125ad5e430d103e7e2420332297c5abdf7aec

**Comment by patriknw** [comment]
unrelated failure in akka-sample-cluster. I'm merging this anyway.

**Pull request merged**